client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx

```diff
@@ -22,37 +22,28 @@ const dateFormat = 'yyyy-MM-dd';
 
 const ClinicalDetails: React.FC = (): JSX.Element => {
     const classes = useStyles();
+    const context = React.useContext(clinicalDetailsDataContext);
 
     const [symptoms, setSymptoms] = React.useState<string[]>([]);
     const [backgroundDiseases, setBackgroundDiseases] = React.useState<string[]>([]);
     const [isUnkonwnDateChecked, setIsUnkonwnDateChecked] = React.useState<boolean>(false);
-    const [hasBackgroundDiseases, setHasBackgroundDiseases] = React.useState<boolean>(false);
+    const [hasBackgroundDiseases, setHasBackgroundDiseases] = React.useState<boolean>(context.clinicalDetailsData.backgroundDeseases.length > 0);
     const [otherSymptom, setOtherSymptom] = React.useState<string>('');
-    const [selectedSymptoms, setSelectedSymptoms] = React.useState<string[]>([]);
-    const [selectedBackgroundDiseases, setSelectedBackgroundDiseases] = React.useState<string[]>([]);
     const [isOtherSymptomChecked, setIsOtherSymptomChecked] = React.useState<boolean>(false);
     const [isOtherBackgroundIllnessChecked, setIsOtherBackgroundIllnessChecked] = React.useState<boolean>(false);
     const [otherBackgroundIllness, setOtherBackgroundIllness] = React.useState<string>('');
     const [isolationCityName, setIsolationCityName] = React.useState<string>('');
     const [isolationStreetName, setIsolationStreetName] = React.useState<string>('');
     const [streetsInCity, setStreetsInCity] = React.useState<Street[]>([]);
 
-    const context = React.useContext(clinicalDetailsDataContext);
+    const showBackgroundDiseases = (hasBackgroundDiseases || context.clinicalDetailsData.backgroundDeseases.length > 0)
```

Comment:
Why do you check them both? In addition, you can use useMemo

Reply:
כבר תוקן